pynest/nest/tests/test_parrot_neuron.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+#
+# test_parrot_neuron.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+# This script tests the parrot_neuron in NEST.
+
+import nest
+import unittest
+
+
+@nest.check_stack
+class ParrotNeuronTestCase(unittest.TestCase):
+    """Check parrot_neuron spike repetition properties"""
+
+    def setUp(self):
+        nest.set_verbosity(100)
+        nest.ResetKernel()
+
+        # set up source spike generator, as well as parrot neurons
+        self.spike_time = 1.
+        self.source = nest.Create("spike_generator", 1, {"spike_times": [self.spike_time]})
+        self.parrot = nest.Create('parrot_neuron')
+        self.spikes = nest.Create("spike_detector")
+
+        # record source and parrot spikes
+        nest.Connect(self.source, self.spikes)
+        nest.Connect(self.parrot, self.spikes)
+
+        self.gid_source = nest.GetStatus(self.source)[0]["global_id"]
+        self.gid_parrot = nest.GetStatus(self.parrot)[0]["global_id"]
+
+    def test_ParrotNeuronRepeatSpike(self):
+        """Check parrot_neuron repeats spikes on port 0"""
+
+        # connect with arbitrary delay
+        delay = .7
+        nest.Connect(self.source, self.parrot, syn_spec={"delay": delay})
+
+        nest.Simulate(3.)
+
+        # get spike from parrot neuron
+        events = nest.GetStatus(self.spikes)[0]["events"]
+        post_time = events['times'][events['senders'] == self.gid_parrot]
+
+        # assert spike was repeated at correct time
+        assert post_time, "Parrot neuron failed to repeat spike."
+        assert self.spike_time + delay == post_time, "Parrot neuron repeated spike at wrong delay"
+
+    def test_ParrotNeuronIgnoreSpike(self):
+        """Check parrot_neuron ignores spikes on port 1"""
+
+        # connect with arbitrary delay to port 1
+        nest.Connect(self.source, self.parrot, syn_spec={"receptor_type": 1})
```

Comment:
I would explicitly specify a delay here as well, just to make it definite that the spike would arrive within the simulated time. Now, you rely on the default delay being 1 ms, but that could in principle change.